hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducer.java

```diff
@@ -577,10 +587,55 @@ public InputStream decompress(InputStream is) {
          *
          * @param blob the blob to publish
          */
+        @Deprecated
         void publish(HollowProducer.Blob blob);
+
+        /**
+         * Publish the blob specified to this publisher's blobstore.
+         * <p>
+         * It is guaranteed that {@code blob} was created by calling one of
+         * {@link BlobStager#openSnapshot(long)}, {@link BlobStager#openDelta(long, long)}, or
+         * {@link BlobStager#openDelta(long, long)} on this publisher, or
+         * {@link BlobStager#openReverseDelta(long, long)} on this publisher, or
+         * {@link BlobStager#openHeader(long)} on this publisher.
+         *
+         * @param publishArtifact the blob to publish
+         */
+        default void publish(HollowProducer.PublishArtifact publishArtifact) {
```

Comment:
Nit: Maybe a comment here that headers are a no-op, and 

since we marked publish(Blob) above deprecated, it'd be good to provide a default implementation (could throw an exception) so that consumers don't have to implement the deprecated method.